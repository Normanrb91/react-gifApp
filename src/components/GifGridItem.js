import React from 'react';
import PropTypes from 'prop-types';


export const GifGridItem = ({title, url}) => {
    return (
        <div className="card text-center">
            <div class="card-body ">
                <img src={url} alt={title} />
                <p className="mt-3">{title}</p>
            </div>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
