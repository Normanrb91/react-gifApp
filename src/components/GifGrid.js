import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {


    const {data:images, loading} = useFetchGifs(category);

    return(
        <>
        <h2 c>{category}</h2>

            {loading && 

                <div class="spinner-border text-success" role="status">
                    <span class="visually-hidden">Espere...</span>
                </div>
            }

            <div className="card-grid mb-2">
            {
                images.map(img =>(
                    <GifGridItem 
                         key={img.id}
                         {... img}
                    />
                ))
            }

        </div>
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
