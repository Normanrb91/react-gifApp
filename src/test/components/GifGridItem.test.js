import React from 'react';
import {shallow} from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';

describe('Pruebas en el componente <GridGifItem />', () => {

    test('Debe mostrar el componente correctamente ', () => {

        const title = 'titulo';
        const url = 'http://localhost/image.jpg'

        const wrapper = shallow(<GifGridItem title={title} url={url}/>)
        expect(wrapper).toMatchSnapshot();
        
    });

})
