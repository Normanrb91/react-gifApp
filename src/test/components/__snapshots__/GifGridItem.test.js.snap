// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GridGifItem /> Debe mostrar el componente correctamente  1`] = `
<div
  className="card"
>
  <img
    alt="titulo"
    src="http://localhost/image.jpg"
  />
  <p>
    titulo
  </p>
</div>
`;
